### DISCUSSION TITLE
ERG spikes

### DISCUSSION THREAD

**Discussion by gurase** [doudar/SmartSpin2k#241] [Q&A]
Not sure if this is due to the firmware my SmartSpin2k auto-updated to (1.8.22-develop) or the last couple Zwift updates, but I've been having a lot of issues with ERG. I'll be in the warmup with an 85 W target, and the resistance just cranks back and forth from 0-100. I have Favero Assioma Uno pedals on an OG Peloton, running Zwift on an Apple TV.

Here's part of my log from my last workout - you can see it going from 0 to 230 W and eventually the resistance was so high the wheel locked. I think at that point Zwift turned off ERG because I wasn't moving.

If it's the firmware, could I get instructions (username/password) for uploading it to the ESP?

Thanks so much!

N 156023 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 156525 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 0 Incline: 159.899994"
N 157024 BLE_Common: "20 00 12 00 c2 00 f7 50 <- 0x1818 | 0x2a63 | CPS[ CD(70.54) PW(18) ]"
N 157027 BLE_Server: "44 02 c4 12 8c 00 12 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(70.00) PW(18) SD(804.00) ]"
N 157029 BLE_Server: "23 00 12 00 00 19 00 8e 59 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(70.00) PW(18) ]"
N 157032 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 157038 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 18 Incline: 173.300003"
N 157517 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 18 Incline: 180.000000"
N 158025 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 18 Incline: 186.699997"
N 158033 BLE_Common: "20 00 1a 00 c3 00 60 54 <- 0x1818 | 0x2a63 | CPS[ CD(70.38) PW(26) ]"
N 158036 BLE_Server: "44 02 c4 12 8c 00 1a 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(70.00) PW(26) SD(804.00) ]"
N 158038 BLE_Server: "23 00 1a 00 00 1a 00 fb 5c -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(70.00) PW(26) ]"
N 158039 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 158530 BLE_Server: "App wrote nothing "
N 158531 BLE_Server: "assuming it's a Control request"
N 159010 BLE_Server: "App wrote nothing "
N 159011 BLE_Server: "assuming it's a Control request"
N 159041 BLE_Common: "20 00 4e 00 c5 00 0e 5b <- 0x1818 | 0x2a63 | CPS[ CD(71.86) PW(78) ]"
N 159045 BLE_Server: "44 02 09 13 8e 00 4e 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(71.00) PW(78) SD(873.00) ]"
N 159049 BLE_Server: "23 00 4e 00 00 1b 00 5c 60 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(71.00) PW(78) ]"
N 159050 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 159585 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 78 Incline: 194.000000"
N 160033 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 78 Incline: 194.699997"
N 160051 BLE_Common: "20 00 42 00 c6 00 4b 5e <- 0x1818 | 0x2a63 | CPS[ CD(74.11) PW(66) ]"
N 160055 BLE_Server: "44 02 d7 13 94 00 42 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(74.00) PW(66) SD(79.00) ]"
N 160057 BLE_Server: "23 00 42 00 00 1c 00 9a 63 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(74.00) PW(66) ]"
N 160058 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 160574 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 66 Incline: 198.500000"
N 161060 BLE_Common: "20 00 5e 00 c7 00 92 61 <- 0x1818 | 0x2a63 | CPS[ CD(73.23) PW(94) ]"
N 161064 BLE_Server: "44 02 92 13 92 00 5e 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(73.00) PW(94) SD(10.00) ]"
N 161068 BLE_Server: "23 00 5e 00 00 1d 00 e3 66 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(73.00) PW(94) ]"
N 161069 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 161086 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 94 Incline: 196.699997"
N 161594 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 94 Incline: 195.800003"
N 162070 BLE_Common: "20 00 72 00 c8 00 e2 64 <- 0x1818 | 0x2a63 | CPS[ CD(72.45) PW(114) ]"
N 162073 BLE_Server: "44 02 4e 13 90 00 72 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(72.00) PW(114) SD(942.00) ]"
N 162076 BLE_Server: "23 00 72 00 00 1e 00 38 6a -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(72.00) PW(114) ]"
N 162077 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 162135 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 114 Incline: 195.279999"
N 162645 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 114 Incline: 195.020004"
N 163080 BLE_Common: "20 00 8c 00 c9 00 3b 68 <- 0x1818 | 0x2a63 | CPS[ CD(71.69) PW(140) ]"
N 163083 BLE_Server: "44 02 09 13 8e 00 8c 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(71.00) PW(140) SD(873.00) ]"
N 163085 BLE_Server: "23 00 8c 00 00 1f 00 99 6d -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(71.00) PW(140) ]"
N 163086 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 163156 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 140 Incline: 194.039993"
N 163690 BLE_Server: "App wrote nothing "
N 163691 BLE_Server: "assuming it's a Control request"
N 164087 BLE_Common: "20 00 e6 00 cb 00 0d 6f <- 0x1818 | 0x2a63 | CPS[ CD(70.38) PW(230) ]"
N 164091 BLE_Server: "44 02 c4 12 8c 00 e6 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(70.00) PW(230) SD(804.00) ]"
N 164093 BLE_Server: "23 00 e6 00 00 20 00 06 71 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(70.00) PW(230) ]"
N 164095 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 164200 BLE_Server: "App wrote nothing "
N 164201 BLE_Server: "assuming it's a Control request"
N 164746 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 230 Incline: 192.660004"
N 165096 BLE_Common: "20 00 fa 00 cc 00 9b 72 <- 0x1818 | 0x2a63 | CPS[ CD(67.52) PW(250) ]"
N 165099 BLE_Server: "44 02 f6 11 86 00 fa 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(67.00) PW(250) SD(598.00) ]"
N 165102 BLE_Server: "23 00 fa 00 00 21 00 9b 74 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(67.00) PW(250) ]"
N 165103 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 165284 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 250 Incline: 191.100006"
N 165794 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 250 Incline: 190.320007"
N 166105 BLE_Common: "20 00 04 01 cd 00 8a 76 <- 0x1818 | 0x2a63 | CPS[ CD(61.01) PW(260) ]"
N 166109 BLE_Server: "44 02 5b 10 7a 00 04 01 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(61.00) PW(260) SD(187.00) ]"
N 166111 BLE_Server: "23 00 04 01 00 22 00 8a 78 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(61.00) PW(260) ]"
N 166112 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 166304 BLE_Server: "05 55 00 -> ERG Mode Target: 85 Current: 260 Incline: 188.660004"
...
N 183240 BLE_Server: "23 00 00 00 00 2c 00 34 b7 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(0.00) PW(0) ]"
N 183242 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 183345 BLE_Server: "05 56 00 -> ERG Mode Target: 86 Current: 0 Incline: 168.050003"
N 183855 BLE_Server: "05 56 00 -> ERG Mode Target: 86 Current: 0 Incline: 168.050003"
N 184243 BLE_Common: "20 00 00 00 d2 00 db 92 <- 0x1818 | 0x2a63 | CPS[ CD(0.00) PW(0) ]"
N 184247 BLE_Server: "44 02 00 00 00 00 00 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(0.00) PW(0) SD(0.00) ]"
N 184248 BLE_Server: "23 00 00 00 00 2c 00 34 b7 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(0.00) PW(0) ]"
N 184251 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 184421 BLE_Server: "App wrote nothing "
N 184422 BLE_Server: "assuming it's a Control request"
N 184932 BLE_Server: "App wrote nothing "
N 184933 BLE_Server: "assuming it's a Control request"
N 185253 BLE_Server: "44 02 00 00 00 00 00 00 00 00 00 00 00 00 -> 0x1826 | 0x2ad2 | FTMS(IBD)[ HR(0) CD(0.00) PW(0) SD(0.00) ]"
N 185255 BLE_Server: "23 00 00 00 00 2c 00 34 b7 -> 0x1826 | 0x2ad2 | CPS(CPM)[ CD(0.00) PW(0) ]"
N 185257 BLE_Server: "00 00 -> 0x180d | 0x2a37 | HRS(HRM)[ HR(0) ]"
N 185358 BLE_Server: "11 00 00 64 00 28 33 -> Sim Mode Incline 1.000000"

**Comment by gurase** [comment]
Update: I figured out how to downgrade to 1.6.19, and that appears to have fixed the ERG issue. The firmware downgrade also changed my shift step, ERG sensitivity, and incline multiplier settings, so I tried 1.8.22 again with those same settings, and had the same issue.

**Reply by doudar** [reply]
Thanks for the info @gurase ! Sorry I'm just seeing this now. 

I'll take a look at the changes and see if I can figure out what might have caused your issue. 

Is it continuing to change the resistance even after you've stopped pedaling or constantly over and under shooting the set point while pedaling? 

If it's the latter, possibly lowering the shift step or ERG sensitivity (1.6.19 may not have that setting) may help - I know the Peloton requires a much lower amount of "knob turn" than other spin bikes.

**Reply by gurase** [reply]
There were a couple times it kept turning the resistance knob to the right even after it was maxed out, and I wasn’t pedaling and actually trying to just quit the workout. But it was definitely over and under shooting too.

I have shift step set to 600 and ERG sensitivity set to 2.8 in 1.6.19 and have no issues with that, other than it being a little slow to react for short intervals (mitigated by shifting early). I tried 600 and 2.8 in 1.8.22 and it was still over/undershooting like crazy.

Really the only issue I have in 1.6.19 is my cadence will occasionally drop for a split second.
